src/components/import/import.js

```diff
@@ -0,0 +1,141 @@
+
+import StarBackground from 'components/star-background';
+import Header, { HeaderType } from 'primitives/text/header';
+import ContentContainer from 'primitives/container/content-container';
+import SubContainer from 'primitives/container/sub-container';
+import LinkIcon from 'primitives/other/link-icon';
+import Button from 'primitives/other/button';
+import LabeledInput from 'primitives/form/labeled-input';
+import Label from 'primitives/form/label';
+
+import { Zap } from 'constants/icons';
+
+import './style.scss';
+
+export default function Import({
+  importSector,
+  updateImport,
+  json,
+  isJsonValid,
+  parsedSectors,
+  sector,
+  intl,
+}) {
+  const updateJson = jsonData => {
+    updateImport('sector', undefined);
+    updateImport('json', jsonData);
+  };
+
+  const updateJsonFromTextarea = ({ target }) => updateJson(target.value);
+
+  const updateJsonFromFile = ({ target }) => {
+    if (target.files.length > 0) {
+      const fileReader = new FileReader();
+      fileReader.onloadend = e => updateJson(e.target.result);
+      fileReader.readAsText(target.files[0]);
+    }
+  };
+
+  const updateSector = ({ value }) => {
+    updateImport('sector', value);
+  };
+
+  const isValid = () => !!sector;
+
+  return (
+    <StarBackground>
+      <ContentContainer direction="column" align="center" justify="center">
+        <Header type={HeaderType.header2}>
+          <FormattedMessage id="misc.import" />
+        </Header>
+        <SubContainer
+          className="Import-Form"
+          noMargin
+          direction="column"
+          align="flexStart"
+        >
+          <Label className="Import-Help">
+            <FormattedMessage id="misc.importHelp" />
+          </Label>
+          <LabeledInput
+            className="Import-JsonFile"
+            isVertical
+            type="file"
+            label="misc.jsonFile"
+            name="jsonFile"
+            accept=".json"
+            onChange={updateJsonFromFile}
+          />
+          <LabeledInput
+            className="Import-JsonInput"
+            isVertical
+            type="textarea"
+            label="misc.jsonData"
+            name="jsonData"
+            value={json}
+            onChange={updateJsonFromTextarea}
+          />
+          <span
+            className={classNames(
+              'Import-JsonValid',
+              isJsonValid ? 'Yes' : 'No',
+            )}
+          >
+            {isJsonValid
+              ? intl.formatMessage({ id: 'misc.importJsonValid' })
+              : intl.formatMessage({ id: 'misc.importJsonInvalid' })}
+          </span>
+          <LabeledInput
+            isVertical
+            type="dropdown"
+            label="misc.selectSectorToImport"
+            placeholder={intl.formatMessage({ id: 'misc.pleaseSelect' })}
+            searchable={false}
+            clearable={false}
+            name="sector"
+            value={sector}
+            options={parsedSectors}
```

Comment:
In what cases would the given sector be parsed into multiple sectors? Instead of a dropdown, should this just be some text confirming valid json and the name of the sector to be imported?

Reply:
The 'sectors' entry in the exported JSON is a array. So I assumed that (maybe in the future) someone could export all their saved sectors at ones or something like that. I just wanted to represent the data structure as close as possible and it allows multiple sectors in one file.

Reply:
I figured a good trade off would be to auto-select a sector if the JSON is valid.
I Implemented that in Irame/sectors-without-number@1b4130f944583959f1a65350d6cfe2c5a7b92b42